### ISSUE TITLE
Logits Shift of Trado Inference

### ISSUE THREAD

**Issue by jianc99** [Gen-Verse/dLLM-RL#18]
Hi there,

Thank you for your amazing work. I just noticed that the output logits of Trado models are the predictions of the next place. For example, if we forward a block with all masks, then the logits of the first place is actually the prediction of the second place. While you directly place the predicted token to the first place without shifting, will this introduce bias?

Thanks
